src/Filtering/ImageGrid/ResampleAnImage/Code.cxx

```diff
@@ -43,49 +43,69 @@ main(int argc, char * argv[])
+
+  /*
+   * We will scale the objects in the image by the factor `scale`; that is they
+   * will be shrunk (scale < 1.0) or enlarged (scale > 1.0).  However, the
+   * number of pixels for each dimension of the output image will equal the
+   * corresponding number of pixels in the input image, with padding (if
+   * shrunk) or cropping (if enlarged) as necessary.  Furthermore, the physical
+   * distance between adjacent pixels will be the same in the input and the
+   * output images.  In contrast, if you want to change the resolution of the
+   * image without changing the represented physical size of the objects in the
+   * image, omit the transform and instead use:
+   *
+   *   outputSize[d] = inputSize[d] * scale;
+   *   outputSpacing[d] = inputSpacing[d] / scale;
+   *   outputOrigin[d] = inputOrigin[d] + 0.5 * (outputSpacing[d] - inputSpacing[d]);
+   *
+   * in the loop over dimensions.
+   */
+
+  typename ImageType::SizeType    outputSize = inputSize;
+  typename ImageType::SpacingType outputSpacing = inputSpacing;
+  // typename ImageType::PointType   outputOrigin = inputOrigin;
 
   using ScaleTransformType = itk::ScaleTransform<ScalarType, Dimension>;
-  ScaleTransformType::Pointer scaleTransform = ScaleTransformType::New();
-
-  ScaleTransformType::ParametersType parameters = scaleTransform->GetParameters();
-  parameters[0] = scale;
-  parameters[1] = scale;
+  typename ScaleTransformType::Pointer scaleTransform = ScaleTransformType::New();
 
-  scaleTransform->SetParameters(parameters);
-  scaleTransform->SetCenter(centralPoint);
+  typename ScaleTransformType::ParametersType scaleTransformParameters = scaleTransform->GetParameters();
+  itk::Point<ScalarType, Dimension>           scaleTransformCenter;
+  for (unsigned int d = 0; d < Dimension; ++d)
+  {
+    scaleTransformParameters[d] = scale;
+    scaleTransformCenter[d] = static_cast<ScalarType>(static_cast<IndexValueType>(inputSize[d] / 2));
+  }
+  scaleTransform->SetParameters(scaleTransformParameters);
+  scaleTransform->SetCenter(scaleTransformCenter);
 
   using LinearInterpolatorType = itk::LinearInterpolateImageFunction<ImageType, ScalarType>;
-  LinearInterpolatorType::Pointer interpolator = LinearInterpolatorType::New();
+  typename LinearInterpolatorType::Pointer interpolator = LinearInterpolatorType::New();
 
   using ResampleFilterType = itk::ResampleImageFilter<ImageType, ImageType>;
-  ResampleFilterType::Pointer resampleFilter = ResampleFilterType::New();
+  typename ResampleFilterType::Pointer resampleFilter = ResampleFilterType::New();
 
   resampleFilter->SetInput(inputImage);
   resampleFilter->SetTransform(scaleTransform);
   resampleFilter->SetInterpolator(interpolator);
-  resampleFilter->SetSize(size);
-  resampleFilter->SetOutputSpacing(spacing);
+  resampleFilter->SetSize(outputSize);
+  resampleFilter->SetOutputSpacing(outputSpacing);
+  // resampleFilter->SetOutputOrigin(outputOrigin);
```

Comment:
`outputOrigin` should be uncommented, because it does not change the output as it is currently defined. When it is active, it participates in testing and is easier to update it if needed in the future.

Reply:
Thank you.  I uncommented all use of origin in both `Code.cxx` and `Code.py` with the latest push.